website/versioned_docs/version-0.63/native-modules-android.md

```diff
@@ -1,116 +1,127 @@
+Add a debug log in the method to confirm it has been invoked when you call it from your application. Below we import and use the [Log](https://developer.android.com/reference/android/util/Log) class from the Android util package like so:
 
 ```java
-  @ReactMethod
-  public void show(String message, int duration) {
-    Toast.makeText(getReactApplicationContext(), message, duration).show();
-  }
+import android.util.Log;
+
+@ReactMethod
+public void createCalendarEvent(String name, String location) {
+   Log.d("CalendarModule", "Create event called with name: " + name
+   + " and location: " + location);
+}
 ```
 
-### Argument Types
+Once we finish implementing the native module and hook it up in JS, you can follow [these steps](https://developer.android.com/studio/debug/am-logcat.html) to view the logs from your app.
+
+### Synchronous Methods
 
-The following argument types are supported for methods annotated with `@ReactMethod` and they directly map to their JavaScript equivalents
+You can pass `isBlockingSynchronousMethod = true` to a native method to mark it as a synchronous method.
 
-```text
-Boolean -> Bool
-Integer -> Number
-Double -> Number
-Float -> Number
-String -> String
-Callback -> function
-ReadableMap -> Object
-ReadableArray -> Array
+```java
+@ReactMethod(isBlockingSynchronousMethod = true)
 ```
 
-Read more about [ReadableMap](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableMap.java) and [ReadableArray](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableArray.java)
+At the moment, we do not recommend this, since calling methods synchronously can have strong performance penalties and introduce threading-related bugs to your native modules. Additionally, please note that if you choose to enable `isBlockingSynchronousMethod`, your app can no longer use the Google Chrome debugger. This is because synchronous methods require the JS VM to share memory with the app. For the Google Chrome debugger, React Native runs inside the JS VM in Google Chrome, and communicates asynchronously with the mobile devices via WebSockets.
 
-### Register the Module
+### Register the Module (Android Specific)
 
-The last step within Java is to register the Module; this happens in the `createNativeModules` of your apps package. If a module is not registered it will not be available from JavaScript.
+Once a native module is written, it needs to be registered with React Native. In order to do so, you need to add your native module to a `ReactPackage` and register the `ReactPackage` with React Native. During initialization, React Native will loop over all packages, and for each `ReactPackage`, register each native module within.
```

Comment:
> register each native module within.

After reading this, I'm left wondering what it means to register a NativeModule with React Native. Maybe we should elaborate on this. What we're really doing is making each NativeModule available to JavaScript code.

**Edit:** Looks like we already talk about this down the line.